hugo/content/docs/ast-types.md

```diff
@@ -3,4 +3,338 @@ title: "AST Type Inference"
 weight: 200
 ---
 
-Coming soon...
+When AST nodes are created during the parsing of a document, they are given a type. The language grammar dictates the shape of those types and how they might be related to each other. There are two ways by which Langium derives AST types from the grammar, by **[inference](#inferred-types)** and by **[declaration](#declared-types)**.
+
+*Inference* is the default behavior in Langium. During the generation of the AST types, Langium infers the possible types directly from the grammar rules. While this a powerful approach for simple languages and prototypes, it is not recommended for more mature languages since minimal changes in the grammar can easily lead to breaking changes.
```

Comment:
> While this a powerful approach

Feels like something is missing here

Reply:
While this **is** a powerful approach...